### PR TITLE
JS Bundling setup

### PR DESCRIPTION
**Pull request by vxsx**
Problems to address

- [x] bundle some js files together
- [x] minify bundled files
- [x] minify not bundled files
- [x] bundle watcher
   - [x] only bundle what's changed / necessary, not on every save

### CODE REVIEW

File: cms/static/cms/js/modules/cms.changelist.js [hunk 1]
```diff
@@ -290,7 +290,7 @@
                             use_inline: true
                             //droppable : ['tree_drop']
                         },
-                        path: false,
+                        path: window.location.origin + '/static/cms/js/jstree/',
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.changelist.js]: static path can be dynamic (it's a django setting)

File: gulpfile.js [hunk 2]
```diff
@@ -43,6 +46,13 @@ var PROJECT_PATTERNS = {
     ]
 };
 
+var JS_BUNDLES = {
```
Comment (FinalAngel) [inline gulpfile.js]: please add some comment describing what it does, how to format

File: cms/static/cms/js/modules/cms.changelist.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.changelist.js]: do we have any helper in js to get the static path?

Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.changelist.js]: not really, in ``toolbar_javascript`` we pass it to ``CMS.config`` via ``{{ STATIC_URL }}``

File: cms/templates/admin/cms/page/tree/base.html [hunk 3]
```diff
@@ -27,15 +27,10 @@
...
-<!-- load changelist -->
-<script src="{% static "cms/js/modules/jquery.ui.custom.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.base.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.changelist.js" %}" type="text/javascript"></script>
-<!-- load jstree -->
-<script src="{% static "cms/js/jstree/_lib/_all.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/jstree/tree_component.js" %}" type="text/javascript"></script>
+{% block jquery %}{% endblock %}
+<script src="{% static "cms/js/dist/bundle.admin.base.min.js" %}" type="text/javascript"></script>
+<!-- load changelist and jstree -->
```
Comment (FinalAngel) [inline cms/templates/admin/cms/page/tree/base.html]: comment here still required?

File: cms/templates/cms/toolbar/toolbar_javascript.html [hunk 4]
```diff
@@ -7,17 +7,7 @@
...
-<script src="{% static "cms/js/modules/jquery.ui.custom.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/jquery.ui.nestedsortable.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.base.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.modal.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.sideframe.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.clipboard.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.plugins.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.structureboard.js" %}" type="text/javascript"></script>
-<script src="{% static "cms/js/modules/cms.toolbar.js" %}" type="text/javascript"></script>
+<script src="{% static "cms/js/dist/bundle.toolbar.min.js" %}" type="text/javascript"></script>
```
Comment (FinalAngel) [inline cms/templates/cms/toolbar/toolbar_javascript.html]: awesome

**Comment by FinalAngel** [comment]
LGTM

File: cms/templates/admin/cms/page/tree/base.html (hunk 3 above)
Comment (vxsx) [inline cms/templates/admin/cms/page/tree/base.html]: well it doesn't hurt

**Pull request merged**